providers/azure/resources/cosmosdb.go

```diff
@@ -42,29 +43,49 @@ func (a *mqlAzureSubscriptionCosmosDbServiceAccount) id() (string, error) {
 func (a *mqlAzureSubscriptionCosmosDbService) accounts() ([]interface{}, error) {
 	conn := a.MqlRuntime.Connection.(*connection.AzureConnection)
 	ctx := context.Background()
-	token := conn.Token()
 	subId := a.SubscriptionId.Data
 
-	accClient, err := cosmosdb.NewDatabaseAccountsClient(subId, token, &arm.ClientOptions{
+	res := []interface{}{}
+
+	// Fetch resources of different types - other than MongoDB and PostgreSQL
+	err := fetchCosmosDBAccounts(ctx, a.MqlRuntime, conn, subId, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = fetchMongoDBClusters(ctx, a.MqlRuntime, conn, subId, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = fetchPostgreSQLServerGroups(ctx, a.MqlRuntime, conn, subId, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func fetchCosmosDBAccounts(ctx context.Context, runtime *plugin.Runtime, conn *connection.AzureConnection, subId string, res *[]interface{}) error {
```

Comment:
```suggestion
func fetchCosmosDBAccounts(ctx context.Context, runtime *plugin.Runtime, conn *connection.AzureConnection, subId string) ([]interface{},error) {
```
we should just return the list here, no need to pass a pointer to it to fill it in

Reply:
OK, I changed overal to return the list. Thanks